packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx

```diff
@@ -0,0 +1,240 @@
+import * as React from 'react';
+import { Box, Collapse, IconButton, Link, styled, Typography } from '@mui/material';
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import ArrowDropDownSharpIcon from '@mui/icons-material/ArrowDropDownSharp';
+import invariant from 'invariant';
+import ComponentCatalogItem from './ComponentCatalogItem';
+import CreateCodeComponentNodeDialog from '../../HierarchyExplorer/CreateCodeComponentNodeDialog';
+import * as appDom from '../../../../appDom';
+import { useDom } from '../../../DomLoader';
+import { usePageEditorApi, usePageEditorState } from '../PageEditorProvider';
+import { useToolpadComponents } from '../../toolpadComponents';
+
+interface FutureComponentSpec {
+  url: string;
+  displayName: string;
```

Comment:
Any reason to change this?
```suggestion
  displayName?: string;
```

Reply:
To avoid "undefined" from being rendered as the display name of a component in the catalog

Reply:
But it used to default to the `id` why remove that?

Reply:
Situations where the `id` and the `displayName` are diverging, for instance, the `id` being "DatePicker" but wanting to show the name of the component as "Date picker"